sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -243,6 +243,7 @@ static const struct dmi_system_id sof_rt5682_quirk_table[] = {
 		.callback = sof_rt5682_quirk_cb,
 		.matches = {
 			DMI_MATCH(DMI_PRODUCT_FAMILY, "Google_Rex"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Rex"),
```

Comment:
Why can't you use DMI_OEM_STRING instead of what looks like a hack?

@cujomalainey what is the direction here?

Reply:
I just found this yesterday as well and had similar questions, I have a call tonight with the ODM to figure out what they are struggling with

Reply:
Also OEM string is actually used by some OEMs so they get upset when we do that. I am trying to avoid using it.

Reply:
Ah I see the issue now, its the BT offload bits :/ We have gotten ourselves into a real pickle with needing to identify endpoints via kernel configs and not firmware :/

Reply:
You could just enable BT offload for all cases where the pin-mux allows for it, and the topology would define if it's used or not.

The rule is that the machine driver exposes a superset of what the topology handles.
The other way around would lead to an error, the topology would use a resource not handled by the machine driver.

Reply:
The downside of that though is then we have to hit udev to flip between BT offload or not to change the topology.

Reply:
Direction forward is to do OEM name hack on new board, not old one until a proper solution is addressed.

Reply:
An alternative would be to package machine driver name into the DMI check in the callback, then you would not see this getting hit as it is not the same machine driver.

Reply:
@plbossart ping here on thoughts as this is blocking ODM

Reply:
I can't follow your last 3 comments @cujomalainey, first there is a direction forward, then an alternative, and now it's blocking - but I am not the person blocking?

we added the DMI_OEM_STRING specifically for Chrome, now it's not good any longer it seems. 
I have no opinion on how you want to deal with your partners, it's beyond the scope of my limited influence. All I can do is review the code for sanity.

Reply:
Ack, I'll stop spinning half baked work arounds and continue using OEM string